src/screen/HomeScreen.js

```diff
@@ -1,77 +1,77 @@
-import React, { useState, useEffect } from 'react';
+import React, {useState, useEffect} from 'react';
 import {
-  ScrollView, View, FlatList, RefreshControl
+    ScrollView, View, FlatList, RefreshControl
 } from 'react-native';
-import { connect } from 'react-redux';
-import { Button } from 'react-native-elements';
-import { ListItem } from '../components/ListItem';
+import {connect} from 'react-redux';
+import {Button} from 'react-native-elements';
+import {ListItem} from '../components/ListItem';
 import Block from '../components/Block';
 import {
-  getPosts, rmPost, savePost, textEdit
+    getPosts, rmPost, savePost, textEdit
 } from '../actions/PostActions';
 
 
 const HomeScreen = ({
-  posts,
-  getPostsAction,
-  navigation,
-}) => {
-  useEffect(() => {
-    getPostsAction();
-  }, []);
+                        posts,
+                        getPostsAction,
+                        navigation,
+                    }) => {
+    componentDidMount()
+    {
+        this.props.getPostsAction();
+    }
+    const [isLoading, setLoading] = useState(false);
 
-  const [isLoading, setLoading] = useState(false);
+    const update = async () => {
+        setLoading(false);
 
-  const update = async () => {
-    setLoading(false);
+        posts = getPostsAction();
+    };
 
-    posts = getPostsAction();
-  };
-
-  const renderItem = ({ item }) => (
-    <ListItem
-      title={item.title}
-      text={item.text}
-      onPress={() => {
-        navigation.navigate('PostScreen', {
-          post: item,
-        });
-      }}
-    />
-  );
-
-  return (
-    <Block height width>
-      <ScrollView
-        refreshControl={(
-          <RefreshControl
-            refreshing={isLoading}
-            onRefresh={update}
-          />
-)}
-      >
-        <FlatList
-          data={posts}
-          renderItem={renderItem}
-          initialNumToRender={10}
-          maxToRenderPerBatch={10}
+    const renderItem = ({item}) => (
+        <ListItem
+            title={item.title}
+            text={item.text}
+            onPress={() => {
+                navigation.navigate('PostScreen', {
```

Comment:
Navigation should be in separate function. PostScreen should be in separate file with NavigationTypes constants.

```
const goTo = (navigationType, params) => navigation.navigate(navigationType, params);
const goToPostScreen = item => goTo('PostScreen', {   post: item });
```